packages/app/src/templates/services/template.service.ts

```diff
@@ -122,26 +126,45 @@ export class TemplateService {
     }
   }
 
-  public async updateTemplateOrder(
-    template: Template
-  ): Promise<RequestResponse<Template>> {
-    const response = await this.deviceService.request({
-      endpoint: Endpoint.Messages,
-      method: Method.Put,
-      body: TemplatePresenter.mapToPureTemplateOrder(template),
+  public async updateTemplatesOrder(
+    templates: Template[]
+  ): Promise<UpdateTemplateOrderRequestResponse> {
+    const results = templates.map(async (template) => {
+      const { status } = await this.deviceService.request({
+        endpoint: Endpoint.Messages,
+        method: Method.Put,
+        body: TemplatePresenter.mapToPureTemplateOrder(template),
+      })
+      return {
+        status,
+        template,
+      }
     })
 
-    if (isResponseSuccess(response)) {
-      this.templateRepository.update(template)
+    const errorTemplates = (await Promise.all(results))
+      .filter(({ status }) => status === RequestResponseStatus.Error)
+      .map(({ template }) => template)
+    const successTemplates = (await Promise.all(results))
+      .filter(({ status }) => status === RequestResponseStatus.Ok)
+      .map(({ template }) => template)
+
+    if (errorTemplates.length > 0) {
+      successTemplates.forEach((template) =>
+        this.templateRepository.update(template)
+      )
 
       return {
-        status: response.status,
-        data: template,
+        status: RequestResponseStatus.Error,
+        error: {
+          message: "Update template order: Something went wrong",
+          data: { errorTemplates, successTemplates },
+        },
       }
     } else {
+      templates.forEach((template) => this.templateRepository.update(template))
+
       return {
-        status: response.status,
-        error: { message: "Update template order: Something went wrong" },
+        status: RequestResponseStatus.Ok,
```

Comment:
Not 100% sure, but wouldn't it be a little simpler version?
```
    successTemplates.forEach((template) =>
      this.templateRepository.update(template)
    )

    if (errorTemplates.length > 0) {
      return {
        status: RequestResponseStatus.Error,
        error: {
          message: "Update template order: Something went wrong",
          data: { errorTemplates, successTemplates },
        },
      }
    } else {
      return {
        status: RequestResponseStatus.Ok,
      }
    }
```

Reply:
Thanks! Changed in [43e06d1](https://github.com/mudita/mudita-center/pull/804/commits/43e06d134aad31f5e351a09965e5574fbf64604f)